packages/frontend-2/components/viewer/Controls.vue

```diff
@@ -396,31 +425,39 @@ const toggleActiveControl = (control: ActiveControl) => {
   activeControl.value = activeControl.value === control ? 'none' : control
 }
 
-onKeyboardShortcut([ModifierKeys.AltOrOpt], 'm', () => {
-  toggleActiveControl('models')
-})
-onKeyboardShortcut([ModifierKeys.AltOrOpt], 'e', () => {
-  toggleActiveControl('explorer')
-})
-onKeyboardShortcut([ModifierKeys.AltOrOpt], 'f', () => {
-  toggleActiveControl('filters')
-})
-onKeyboardShortcut([ModifierKeys.AltOrOpt], ['t'], () => {
-  toggleActiveControl('discussions')
-})
-onKeyboardShortcut([ModifierKeys.AltOrOpt], 'd', () => {
-  toggleActiveControl('measurements')
-})
+const handleKeyboardAction = (action: ViewerKeyboardActions) => {
+  switch (action) {
+    case ViewerKeyboardActions.ToggleModels:
+      toggleActiveControl('models')
+      break
+    case ViewerKeyboardActions.ToggleExplorer:
+      toggleActiveControl('explorer')
+      break
+    case ViewerKeyboardActions.ToggleDiscussions:
+      toggleActiveControl('discussions')
+      break
+    case ViewerKeyboardActions.ToggleMeasurements:
+      toggleMeasurements()
+      break
+    case ViewerKeyboardActions.ToggleProjection:
+      trackAndtoggleProjection()
+      break
+    case ViewerKeyboardActions.ToggleSectionBox:
+      toggleSectionBox()
+      break
+    case ViewerKeyboardActions.ZoomExtentsOrSelection:
+      trackAndzoomExtentsOrSelection()
+      break
+  }
+}
 
-// Viewer actions kbd shortcuts
-onKeyboardShortcut([ModifierKeys.AltOrOpt], ' ', () => {
-  trackAndzoomExtentsOrSelection()
-})
-onKeyboardShortcut([ModifierKeys.AltOrOpt], 'p', () => {
-  toggleProjection()
-})
-onKeyboardShortcut([ModifierKeys.AltOrOpt], 'b', () => {
-  toggleSectionBox()
+Object.keys(map).forEach((actionKey) => {
+  const action = parseInt(actionKey) as ViewerKeyboardActions
```

Comment:
Why the parseInt? The indices should already be numbers.

If TS is saying that they might not be, I think it could be cause in runtime TS enums get converted to an object that has not only the key->value pairs you've specified, but also the inverse: value->key pairs. So it could be that you can also get each enum name as a key. If that's true (validate this), parseInt() is def not what you want to do and probably would trigger some buggy behaviour. You should instead filter those string keys out.